src/base/hip/hip_utils.hpp

```diff
@@ -206,7 +206,7 @@ namespace rocalution
 
     __device__ int __llvm_amdgcn_readlane(int index, int offset) __asm("llvm.amdgcn.readlane");
 
-#ifndef __gfx1030__
+#if !defined(__gfx1030__) && !defined(__gfx1031__)
```

Comment:
As per @cgmb suggestion, I think we probably want:

#ifndef ROCSPARSE_USE_MOVE_DPP
#if defined(__gfx803__) || \
    defined(__gfx900__) || \
    defined(__gfx906__) || \
    defined(__gfx908__) || \
    defined(__gfx90a__)
#define ROCSPARSE_USE_MOVE_DPP 1
#else
#define ROCSPARSE_USE_MOVE_DPP 0
#endif
#endif

At the top and then you can just do:

#if ROCSPARSE_USE_MOVE_DPP

here.

Reply:
We may want to add some other GPUs such as "gfx1010, "gfx1012", etc. since this is also blocking the build for those others.

something similar to https://github.com/xuhuisheng/rocm-build/blob/master/patch/25.rocsparse-gfx10-1.patch

Reply:
@kreigiron, I am the original author of the patch you linked. That is an earlier version of the change that @jsandham and I mentioned.

For the final version that got merged into rocSPARSE, I inverted the if/else condition to maximize compatibility with future architectures. In the updated approach, any architectures not specifically listed in the check will avoid using `__hip_move_dpp`. So, because gfx1010 and gfx1012 are not listed, they will work correctly.

Reply:
that's great, hope this patch gets released soon, it's the only issue that I found so far when compiling from sources for my gpu (gfx1012), it compiled and it's working ok so far.